cpu/samd21/periph/i2c.c

```diff
@@ -197,7 +197,7 @@ int i2c_init_master(i2c_t dev, i2c_speed_t speed)
     i2c_poweron(dev);
 
     /* Start timeout if bus state is unknown. */
-    while (!(I2CSercom->STATUS.reg & SERCOM_I2CM_STATUS_BUSSTATE(1))) {
+    while (!(I2CSercom->STATUS.reg & SERCOM_I2CM_STATUS_BUSSTATE_Msk)) {
```

Comment:
Unknown is 0b00, we only want to time out if the bus state is equal to this. In the old code, if the bus state was Owner (0b10) we would still time out and override it to idle. This is an unlikely state anyway though, to be fair.